src/seedu/addressbook/AddressBook.java

```diff
@@ -893,19 +931,148 @@ private static String encodePersonToString(String[] person) {
      * the return value may not always be present.
      * ====================================================================
      */
+    
+    /**
+     * Decodes edit data from it's supposed string representation.
+     *
+     * @param encoded string to be decoded
+     * @return if cannot decode: empty Optional
+     *         else: Optional containing decoded person
+     */
+    private static Optional<ArrayList<String>> decodeEditDataFromString(String encoded) {
+        // check that we can extract index and at least 1 property to be edited from the encoded string
+        if (!isEditDataExtractableFrom(encoded)) { 
+            return Optional.empty();
+        }
+        final ArrayList<String> decodedData = getEditDataFromString(encoded);
+        // check that the edit data is valid
+        return isEditDataValid(decodedData) ? Optional.of(decodedData) : Optional.empty();
+    }
+    
     /**
+     * Decodes edit data from it's supposed string representation.
+     *
+     * @param encoded string to be decoded
+     * @return ArrayList<String> containing index, and/or name, phone, email
+     */
+    private static ArrayList<String> getEditDataFromString(String encoded) {
+		ArrayList<String> decodedData = new ArrayList<>();
+		encoded = encoded.trim();
+		int indexOfWhitespace = encoded.indexOf(' ');
+		decodedData.add( new String(encoded.substring(0, indexOfWhitespace))); //Extracts targetIndex from string
+		getPersonPropertyFromString(encoded.substring(indexOfWhitespace).trim(), decodedData);		
+		return decodedData;
+	}
+
+	/**
+	 * Assigns the corresponding properties (or null if it does not exist)
+	 * to be edited to ArrayList<String>
+	 * 
+	 * @param encoded
+	 * @param decodedData
+	 */
+	private static void getPersonPropertyFromString(String encoded, ArrayList<String> decodedData) {
+		decodedData.add(getNameIfExist(encoded));
+		decodedData.add(getPhoneIfExist(encoded));
+		decodedData.add(getEmailIfExist(encoded));
+	}
+
+    /**
+     * Decodes name from encoded string representation if it exists.
+     *
+     * @param encoded string to be decoded
+     * @return if n/ exist, return name in string
+     *         else return null
+     */
+	private static String getNameIfExist(String encoded) {
+		if (!encoded.contains(PERSON_DATA_PREFIX_NAME))
+			return null;
+		else {
+			int startIndex = encoded.indexOf(PERSON_DATA_PREFIX_NAME) + PERSON_DATA_PREFIX_NAME.length();
+			String trimmed = encoded.substring(startIndex).trim();
+			String extractedName;
+			if (trimmed.contains("/"))
+				extractedName = trimmed.substring(0, trimmed.indexOf("/")-1).trim();
+			else 
+				extractedName = trimmed;
+			return extractedName;
```

Comment:
Indentation problem here?. Use spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.